tree/dataframe/src/RDFUtils.cxx

```diff
@@ -202,10 +202,15 @@ std::string GetBranchOrLeafTypeName(TTree &t, const std::string &colName)
             }
             return be->GetClassName();
          }
+      } else if (branch->IsA() == TBranch::Class() && branch->GetListOfLeaves()->GetEntries() == 1) {
+         // normal branch (not a TBranchElement): if it has only one leaf, we pick the type of the leaf:
+         // RDF and TTreeReader allow referring to branch.leaf as just branch if branch has only one leaf
+         leaf = static_cast<TLeaf *>(branch->GetListOfLeaves()->UncheckedAt(0));
+         return GetLeafTypeName(leaf, colName + '.' + leaf->GetName());
```

Comment:
```suggestion
         return GetLeafTypeName(leaf, leaf->GetFullName());
```

Reply:
Thanks, I will change this for all occurrences of `GetLeafTypeName` in another PR